### PR TITLE
[JAX] Rework amax reduction over TPSP

### PR DESCRIPTION
**Pull request by phu0ngng**
# Description

In this PR, we add an additional check to detect whether the input is partitioned in the sequence dimension and only trigger the amax reduction across TPSP in that case.
With this change, the `using_global_amax_of_x` is no longer needed.

## Type of change

- [ ] Documentation change (change only to the documentation, either a fix or a new content)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Infra/Build change
- [x] Code refactoring

# Checklist:

- [x] I have read and followed the [contributing guidelines](https://github.com/NVIDIA/TransformerEngine/blob/main/CONTRIBUTING.rst)
- [x] The functionality is complete
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: transformer_engine/jax/cpp_extensions/quantization.py [hunk 1]
```diff
@@ -613,25 +620,26 @@ def partition(
...
+                batch_sequence_transpose=batch_sequence_transpose,
             )
-            if amax_scope is AmaxScope.TPSP:  # Run AR across TP/SP
-                gmesh = global_mesh_resource()
-                amax = lax_paral_op(amax, jax.lax.pmax, gmesh.tp_resource, mesh)
+            gmesh = global_mesh_resource()
+            sequence_dim = 0 if batch_sequence_transpose else 1
+            # Run AR across TPSP only when tensor-sequence is detected in the input spec
+            if amax_scope is AmaxScope.TPSP and x_spec[sequence_dim] == gmesh.tpsp_resource:
                 amax = lax_paral_op(amax, jax.lax.pmax, gmesh.tpsp_resource, mesh)
```
Comment (jberchtold-nvidia) [inline transformer_engine/jax/cpp_extensions/quantization.py:640]: Previously, we would also AR in the `gmesh.tp_resource` dim. To keep the previous behavior, do we still need to reduce on `TP` if `amax_scope is TPSP` but the sequence dim is `TP` not `TPSP`? Or is it okay to only reduce when `tpsp` is active?

**Review by jberchtold-nvidia — APPROVED** [review]
Left one question, pending that LGTM. Thanks!

File: transformer_engine/jax/cpp_extensions/quantization.py (hunk 1 above)
Comment (phu0ngng) [reply transformer_engine/jax/cpp_extensions/quantization.py:640]: We only do AG the sequence dimension before the GEMM in TPSP, thus amax reduction is needed.

If users use TP for TPSP, there are existing warnings in the GEMM op to warn users to switch to TPSP.

**Comment by phu0ngng** [comment]
pipeline 35814193 passed.
